### ISSUE TITLE
automatically keep up-to-date movie ratings

### ISSUE THREAD

**Issue by greybax** [greybax/awesome-IT-films#23]
Need to investigate some way to keep movie ratings up-to-date automatically in accordance with IMDb?
Ex. https://github.com/worr/node-imdb-api

**Comment by perkinsjr** [comment]
We do have the potential option of instead of using the raw Markdown files could optionally do a call to the API get all the film data into a json and then convert the json to Markdown

imdb.get({name: 'The Matrix'}, {apiKey: 'APIKEY', timeout: 30000}).then( ...)

{
   "Title":"Matrix",
   "Year":"1993–",
   "Rated":"N/A",
   "Released":"01 Mar 1993",
   "Runtime":"60 min",
   "Genre":"Action, Drama, Fantasy, Thriller",
   "Director":"N/A",
   ..... "MORE DATA"
}

Or we could do the above only to the single field for ImdbRating 

Thoughts? @greybax

**Comment by greybax** [comment]
@perkinsjr sounds good! 👍 I think for now will be enough ImdbRating field, but I hope we can add more fields later as we need them.

**Comment by perkinsjr** [comment]
Sounds like a plan, I will take a look after I finish some work up.

**Comment by perkinsjr** [comment]
After looking at what Markdown is there, I feel like it would be quicker to rewrite the way MD is created, we can create a JSON with all imdb id's in and then just do the following: 

```
const imdb = require('imdb-api')
const fs = require('fs')
const json2md = require('json2md')


imdb.get({id: 'tt0133093'}, {apiKey: 'c6691945', timeout: 30000}).then((movie) => {
    console.log(json2md([
        { h4:  movie.title},
        {blockquote: movie.plot},
        { img: { title:  movie.title, source: movie.poster } },
        { ul: ["year: " + movie.year , "Country: UK, USA", { link: { title: "imdb", source:movie.imdburl }}]}
        ]));
});
```

This will return:
#### The Matrix

> Thomas A. Anderson is a man living two lives. By day he is an average computer programmer and by night a hacker known as Neo. Neo has always questioned his reality, but the truth is far beyond his imagination. Neo finds himself targeted by the police when he is contacted by Morpheus, a legendary computer hacker branded a terrorist by the government. Morpheus awakens Neo to the real world, a ravaged wasteland where most of humanity have been captured by a race of machines that live off of the humans' body heat and electrochemical energy and who imprison their minds within an artificial reality known as the Matrix. As a rebel against the machines, Neo must return to the Matrix and confront the agents: super-powerful computer programs devoted to snuffing out Neo and the entire human rebellion.

![The Matrix](https://m.media-amazon.com/images/M/MV5BNzQzOTk3OTAtNDQ0Zi00ZTVkLWI0MTEtMDllZjNkYzNjNTc4L2ltYWdlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_SX300.jpg)


 - year: 1999
 - Country: UK, USA
 - [imdb](https://www.imdb.com/title/tt0133093)

And then we can sort by rating..

Just a thought..

 I can go back to original idea but I think it would make more sense. Thus when a new movie is added we can just add it to the json re-run the javascript and we have a new movie / tv / documentary added

**Comment by greybax** [comment]
This might be a good solution. but in this case we will have a strong dependecy on `imdb-api`, which is not good I think. Because almost all content in markdown we will generate based on data from this api.
What if IMDb API will change or api policy will change?

**Comment by perkinsjr** [comment]
That is a fair point, I will look to see if there is an elegant solution to just the ratings with minimal exposure to the API. Then in the future if you decide you want more of driven API direction we can circle around.

**Comment by perkinsjr** [comment]
This solution is fairly elegant, and should work on both a level of keeping the API separate and making sure that we don't have an issue: 

Firstly we should create a helper js file that we can export a JSON that contains key and values like so:

```
exports.MOVIES = {
    "Matrix_Score": "tt0133093"
}
```

Then a file to handle the api calls:

```
const imdb = require('imdb-api')
const fs = require('fs')
const helper  = require('./helper')


imdb.get({id: helper.MOVIES['Matrix_Score']}, {apiKey: 'c6691945', timeout: 30000}).then((movie) => {
    fs.readFile('./movies_prebuilt.txt', 'utf8', function (err,data) {
        if (err) {
          return console.log(err);
        }
        var result = data.replace(/Matrix_Score/g, movie.rating);
      
        fs.writeFile('./movies.md', result, 'utf8', function (err) {
           if (err) return console.log(err);
        });
      });
});
```

And a seperate file called movies_prebuilt.txt which contains all the Markdown with variables for ratings: 

### Matrix_Score

#### Matrix
> A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.


When the JS is ran before deploying it will update all the ratings to the latest from IMDB. If the updating fails, it will keep the old one and therefore not causing an issue. 

When a new movie / tv / doc comes in we can add it to the prebuilt.txt and the helper.js and run the builder which will update the markdown, get new scores and then we can check in the newest md.

Let me know if you have any changes or issues with the proposal. If not the level of effort is fairly minimal

**Comment by greybax** [comment]
Good, I think we can go with it. Just One think let's rename `helper.js` to `db.js` or `storage.js` or `score.js` (seems that this file will just store only movie scores..)

**Comment by greybax** [comment]
Fixed see PR #63
